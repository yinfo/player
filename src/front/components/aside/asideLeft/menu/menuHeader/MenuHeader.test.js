// @flow

import React from 'react';
import { shallow, mount } from 'enzyme';
import MenuHeader from './MenuHeader';

describe('MenuHeader component', () => {
  it('renders as expected', () => {
    const onClick = jest.fn();

    const component = shallow(
      <div>
        <MenuHeader
          title="menuHeaderTitle"
          backColor="#F1F2F3"
          isCollapsed
          onClick={onClick}
        />
      </div>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const onClick = jest.fn();

    const component = mount(
      <div>
        <MenuHeader
          title="menuHeaderTitle"
          backColor="#F1F2F3"
          isCollapsed
          onClick={onClick}
        />
      </div>,
    );
    const button = component.find('a').first();
    const icon = component.find('.fa-angle-up').first();

    // console.log('icon: ', icon.props().style);
    // expect(icon.props().style).toMatchObject({
    //   WebkitTransform: 'rotate(180deg)',
    //   transform: 'rotate(180deg)',
    // });
    button.simulate('click');
    expect(onClick).toHaveBeenCalled();
    expect(icon.props().style).toMatchObject({
      WebkitTransform: 'rotate(0deg)',
      transform: 'rotate(0deg)',
    });
  });

  it('should have icon 180 deg rotation', () => {
    const onClick = jest.fn();

    const component = mount(
      <MenuHeader
        title="menuHeaderTitle"
        backColor="#F1F2F3"
        isCollapsed={false}
        onClick={onClick}
      />,
    );
    const icon = component.find('.fa-angle-up').first();

    expect(icon.props().style).toMatchObject({
      WebkitTransform: 'rotate(180deg)',
      transform: 'rotate(180deg)',
    });
  });

  it('should have icon 0 deg rotation', () => {
    const onClick = jest.fn();

    const component = mount(
      <MenuHeader
        title="menuHeaderTitle"
        backColor="#F1F2F3"
        isCollapsed={true}
        onClick={onClick}
      />,
    );
    const icon = component.find('.fa-angle-up').first();

    expect(icon.props().style).toMatchObject({
      WebkitTransform: 'rotate(0deg)',
      transform: 'rotate(0deg)',
    });
  });
});
