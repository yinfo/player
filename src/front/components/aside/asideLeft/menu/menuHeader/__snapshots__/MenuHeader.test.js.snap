// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuHeader component renders as expected 1`] = `ShallowWrapper {}`;
